extensions/wikia/Recirculation/services/DiscussionsDataService.class.php

```diff
@@ -42,10 +46,38 @@ function() {
 	 * Get posts for discussions
 	 * @return an array of posts
 	 */
-	public function getPosts() {
-		$data = $this->getData();
+	private function getPosts( $rawData ) {
+		$data = [
+			'posts' => []
+		];
+
+		$rawPosts = $rawData['_embedded']['doc:threads'];
+
+		if ( is_array( $rawPosts ) && count( $rawPosts ) > 0 ) {
+			foreach ( $rawPosts as $key => $value ) {
+				$data['posts'][] = $this->buildPost( $value, $key );
+			}
+		}
 
-		return $data['posts'];
+		return $data;
+	}
+
+	private function formatData( $rawData ) {
+		$data = [];
+		$siteId = $rawData['siteId'];
+
+		$rawPosts = $rawData['_embedded']['doc:threads'];
+		$data['discussionsUrl'] = $this->server . '/d/f';
+		$data['postCount'] = $rawData['threadCount'];
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "postCount" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)